src/components/Accordion/types.ts

```diff
@@ -17,14 +17,14 @@ export type AriaAccordionItemProps = {
     header: Omit<AccordionHeaderProps, 'isOpen'> & { active?: boolean; onClick?: () => void };
     padding?: boolean;
     divider?: boolean;
-    headerComponent?: (props: AccordionHeaderProps) => ReactElement;
+    headerComponent?: (props: AccordionHeaderProps) => ReactElement | null;
 };
 
 export type AccordionItemProps = PropsWithChildren<{
     header: Omit<AccordionHeaderProps, 'isOpen'> & { active?: boolean; onClick?: () => void };
     padding?: boolean;
     divider?: boolean;
-    headerComponent?: (props: AccordionHeaderProps) => ReactElement;
+    headerComponent?: (props: AccordionHeaderProps) => ReactElement | null;
```

Comment:
Not needed anymore

Reply:
You can update the Clarify code to comply, Fondue shouldn't adapt to Clarify (this is important)

Reply:
This is the same discussion we are having here: https://github.com/Frontify/fondue/pull/1340
I'm using `JSX.Element` which is the default inferred type by Typescript when you create a component without `React.FC`

Reply:
Changes on Clarify in this PR https://github.com/Frontify/clarify/pull/14480

Reply:
Fair point, requires those changes to be released first
https://github.com/Frontify/clarify/pull/14562

Reply:
removed them 👍